File: stdin_primer/ruby/05-04_integer_space_step4.rb
# 10 個の整数の半角スペース区切りの入力 (paizaランク D 相当)
# https://paiza.jp/works/mondai/stdin_primer/stdin_primer__integer_space_step4

# 入出力例
INPUT1 = <<~"EOS"
  10 9 8 7 6 5 4 3 2 1
EOS
OUTPUT1 = <<~"EOS"
  10
  9
  8
  7
  6
  5
  4
  3
  2
  1
EOS

# 解答例
# 入力
ary = gets.split.map(&:to_i)
# 出力
puts ary

File: stdin_primer/haskell/05-04_integer_space_step4.hs
-- 10 個の整数の半角スペース区切りの入力 (paizaランク D 相当)
-- https://paiza.jp/works/mondai/stdin_primer/stdin_primer__integer_space_step4

toInt :: String -> Int
toInt x = read x :: Int

toStr :: Int -> String
toStr x = show x :: String

main :: IO ()
main = do
  inputStr <- getLine

  let strNums = words inputStr
  let intNums = map toInt strNums
  let outputStr = unlines $ map toStr intNums

  putStr outputStr
